libraries/AP_Proximity/AP_Proximity_TeraRangerTower.cpp

```diff
@@ -62,8 +62,11 @@ bool AP_Proximity_TeraRangerTower::read_sensor_data()
     int16_t nbytes = _uart->available();
 
     while (nbytes-- > 0) {
-        char c = _uart->read();
-        if (c == 'T' ) {
+        int16_t c = _uart->read();
+        if (c==-1) {
+            return false;
+        }
+        else if (static_cast<char>(c) == 'T' ) {
```

Comment:
I think this else can be removed because of the return before it.  Same for the TeraRangerTowerEvo.cpp changes as well. txs